assets/app/scripts/controllers/projects.js

```diff
@@ -24,6 +24,45 @@ angular.module('openshiftConsole')
       $('#openshift-logo').off('click.projectsPage');
     });
 
+    $scope.toggleDelete = function(project) {
+      // to toggle prompt to delete
+      // disable the delete button if current project is being deleted
+      if (!$scope.projects[project]['deleting']) {
+        $scope.projects[project]['toggled'] = !$scope.projects[project]['toggled'];
+      }
+    };
+
+    $scope.deleteProject = function(project) {
+      $scope.alerts = {};
+      $scope.projects[project]['toggled'] = false;
+      $scope.projects[project]['deleting'] = true;
+      DataService.delete('projects', project, $scope)
+      .then(function() {
+        $scope.alerts = [{
+          type: "success",
+          message: "Project " + $scope.projects[project]['metadata']['annotations']['openshift.io/display-name'] + " was successfully deleted."
+        }];
+        delete $scope.projects[project];
+        $timeout(function() {
+          $scope.alerts = {};
+        }, 10000);
+      }, function(result) {
```

Comment:
hmm, agree there is inconsistency. mostly for readability, I guess... I find this style easier to see the normal and error-handling blocks:
```
foo
.then(function(){
  ...
})
.catch(function(){
  ...
})
```